tutorials/oauth/python/consumer.py

```diff
@@ -7,63 +7,115 @@
+from pprint import pformat
+
+
+def stats_cb(stats_json_str):
+    stats_json = json.loads(stats_json_str)
+    print('\nKAFKA Stats: {}\n'.format(pformat(stats_json)))
+
+
+def oauth_cb(cred, namespace_fqdn, config):
+    access_token = cred.get_token('https://%s/.default' % namespace_fqdn)
+    return access_token.token, access_token.expires_on
+
+
+def print_usage_and_exit(program_name):
+    sys.stderr.write(
+        'Usage: %s [options..] <eventhubs-namespace> <group> <topic1> <topic2> ..\n' % program_name)
+    options = '''
+ Options:
+  -T <intvl>   Enable client statistics at specified interval (ms)
+'''
+    sys.stderr.write(options)
+    sys.exit(1)
+
+
+if __name__ == '__main__':
+    optlist, argv = getopt.getopt(sys.argv[1:], 'T:')
+    if len(argv) < 3:
+        print_usage_and_exit(sys.argv[0])
+
+    namespace = argv[0]
+    group = argv[1]
+    topics = argv[2:]
+
+    # Azure credential
+    cred = DefaultAzureCredential()
+
+    # Consumer configuration
+    # See https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md
+    conf = {
+        'bootstrap.servers': '%s:9093' % namespace,
+        'group.id': group,
+        'session.timeout.ms': 6000,
+        'auto.offset.reset': 'earliest',
+
+        # Required OAuth2 configuration properties
+        'security.protocol': 'SASL_SSL',
+        'sasl.mechanism': 'OAUTHBEARER',
+        'oauth_cb': partial(oauth_cb, cred, namespace),
+    }
+
+    # Check to see if -T option exists
+    for opt in optlist:
+        if opt[0] != '-T':
+            continue
+        try:
+            intval = int(opt[1])
+        except ValueError:
+            sys.stderr.write("Invalid option value for -T: %s\n" % opt[1])
+            sys.exit(1)
+
+        if intval <= 0:
+            sys.stderr.write("-T option value needs to be larger than zero: %s\n" % opt[1])
+            sys.exit(1)
+
+        conf['stats_cb'] = stats_cb
+        conf['statistics.interval.ms'] = int(opt[1])
+
+    # Create logger for consumer (logs will be emitted when poll() is called)
+    logger = logging.getLogger('consumer')
+    logger.setLevel(logging.DEBUG)
+    handler = logging.StreamHandler()
+    handler.setFormatter(logging.Formatter('%(asctime)-15s %(levelname)-8s %(message)s'))
+    logger.addHandler(handler)
+
+    # Create Consumer instance
+    # Hint: try debug='fetch' to generate some log messages
+    c = Consumer(conf, logger=logger)
+
+    def print_assignment(consumer, partitions):
+        print('Assignment:', partitions)
+
+    # Subscribe to topics
+    c.subscribe(topics, on_assign=print_assignment)
+
+    # Read messages from Kafka, print to stdout
+    try:
+        while True:
+            msg = c.poll(timeout=1.0)
+            if msg is None:
+                continue
+            if msg.error():
+                raise KafkaException(msg.error())
+            else:
+                # Proper message
+                sys.stderr.write('%% %s [%d] at offset %d with key %s:\n' %
```

Comment:
Should this be writing to stdout?
Also not sure if this was intentional but this prints a % at the beginning of the line
ex:
```
% mytopic [0] at offset 136 with key None:
b'0'
```

Reply:
Yeah I might do it differently if this was all our code from scratch

It made sense to me that the priority for user-facing samples should be

* they are correct
* they look/feel consistent (at least within one repo)
* <maybe some other things>
* use the "best" Python code style

Nearly all of this is lifted directly from https://github.com/confluentinc/confluent-kafka-python/blob/master/examples/consumer.py and/or https://github.com/Azure/azure-event-hubs-for-kafka/blob/master/quickstart/python/consumer.py

Obv we can change anything later